src/System.Net.Http/tests/FunctionalTests/HttpClient.SelectedSitesTest.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public abstract class HttpClient_SelectedSites_Test : HttpClientTestBase
+    {
+        [Theory]
+        [OuterLoop]
+        [Trait("SelectedSites", "true")]
+        [MemberData(nameof(GetSelectedSites))]
+        public async Task RetrieveSite_Succeeds(string site)
+        {
+            // Not doing this in bulk for platform handlers.
+            if (!UseSocketsHttpHandler)
+                return;
+
+            int remainingAttempts = 2;
+            while (remainingAttempts-- > 0)
+            {
+                try
+                {
+                    await VisitSite(site);
+                    return;
+                }
+                catch
+                {
+                    if (remainingAttempts < 1)
+                        throw;
+                    await Task.Delay(1500);
+                }
+            }
+
+            throw new Exception("Not expected to reach here");
+        }
+
+        [Theory]
+        [OuterLoop]
+        [Trait("SiteInvestigation", "true")]
+        [InlineData("http://microsoft.com")]
+        public async Task RetrieveSite_Debug_Helper(string site)
+        {
+            await VisitSite(site);
+        }
+
+        public static IEnumerable<string[]> GetSelectedSites()
+        {
+            const string resourceName = "SelectedSitesTest.txt";
+            Assembly assembly = typeof(HttpClient_SelectedSites_Test).Assembly;
+            Stream s = assembly.GetManifestResourceStream(resourceName);
+            if (s == null)
+            {
+                throw new Exception("Couldn't find resource " + resourceName);
+            }
+
+            using (var reader = new StreamReader(s))
+            {
+                string site;
+                while (null != (site = reader.ReadLine()))
+                {
+                    yield return new[] { site };
+                }
+            }
+        }
+
+        private async Task VisitSite(string site)
+        {
+            using (HttpClient httpClient = CreateHttpClientForSiteVisit())
```

Comment:
Do we need to create a new HttpClient per site?

Reply:
No, I had some runs with a single HttpClient and went back and forth between it and creating a new one. I didn't notice any relevant difference, if you think it is worth I can change to a single HttpClient.

Reply:
Either way is fine.  We should be able to reuse it, so I was curious why a new one was being used each time.  What you have is fine, though.